src/units/unit.cpp

```diff
@@ -1993,17 +1993,15 @@ std::string unit::describe_builtin_effect(const std::string& apply_to, const con
 	if(apply_to == "attack") {
 		std::vector<t_string> attack_names;
 
-		std::string desc;
-		for(attack_ptr a : attacks_) {
-			bool affected = a->describe_modification(effect, &desc);
-			if(affected && !desc.empty()) {
+		for(const attack_ptr& a : attacks_) {
+			if(a->matches_filter(effect)) {
```

Comment:
Why is the behavior change a good idea?

Reply:
Sure, but the description is not checked for empty anymore. The loop should be skipped entirely if the description is empty.